src/libraries/System.IO.Compression/src/System/IO/Compression/GZipStream.cs

```diff
@@ -7,82 +7,220 @@
+        /// <param name="offset">The byte offset in <paramref name="buffer" /> at which to begin reading data from the stream.</param>
+        /// <param name="count">The maximum number of bytes to read.</param>
+        /// <param name="asyncCallback">An optional asynchronous callback, to be called when the read operation is complete.</param>
+        /// <param name="asyncState">A user-provided object that distinguishes this particular asynchronous read request from other requests.</param>
+        /// <returns>An object that represents the asynchronous read operation, which could still be pending.</returns>
+        /// <remarks><format type="text/markdown"><![CDATA[
+        /// [!INCLUDE[remarks](~/includes/remarks/System.IO.Compression/GZipStream/BeginRead_Byte_Int32_Int32_AsyncCallback_Object.md)]
+        /// ]]></format></remarks>
+        /// <exception cref="System.IO.IOException">The method tried to  read asynchronously past the end of the stream, or a disk error occurred.</exception>
+        /// <exception cref="System.ArgumentException">One or more of the arguments is invalid.</exception>
+        /// <exception cref="System.ObjectDisposedException">Methods were called after the stream was closed.</exception>
+        /// <exception cref="System.NotSupportedException">The current <see cref="System.IO.Compression.GZipStream" /> implementation does not support the read operation.</exception>
+        /// <exception cref="System.InvalidOperationException">A read operation cannot be performed because the stream is closed.</exception>
         public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback? asyncCallback, object? asyncState) =>
             TaskToApm.Begin(ReadAsync(buffer, offset, count, CancellationToken.None), asyncCallback, asyncState);
 
+        /// <summary>Waits for the pending asynchronous read to complete. (Consider using the <see cref="System.IO.Stream.ReadAsync(byte[],int,int)" /> method instead.)</summary>
+        /// <param name="asyncResult">The reference to the pending asynchronous request to finish.</param>
+        /// <returns>The number of bytes read from the stream, between 0 (zero) and the number of bytes you requested. <see cref="System.IO.Compression.GZipStream" /> returns 0 only at the end of the stream; otherwise, it blocks until at least one byte is available.</returns>
+        /// <remarks>Starting with the .NET Framework 4.5, you can perform asynchronous read operations by using the <see cref="O:System.IO.Stream.ReadAsync" /> method. The <see cref="System.IO.Compression.GZipStream.EndRead" /> method is still available in .NET Framework 4.5 to support legacy code; however, you can implement asynchronous I/O operations more easily by using the new async methods. For more information, see <a href="https://docs.microsoft.com/en-us/dotnet/standard/io/asynchronous-file-i-o">Asynchronous File I/O</a>.
```

Comment:
Why does the `O:` in `<see cref="O:System.IO.Stream.ReadAsync" />` mean?

Reply:
It is for referencing an overloaded method. [Notes here](https://github.com/carlossanlop/DocsPortingTool/blob/58a8936f1c7960baf48d6d148b230239a2cc5e68/BackportInstructions.md#user-content-important:~:text=MS%20Docs%20supports%20using%20%3Csee%20cref%3D%22DocId%22%20%2F%3Eelements%20that%20point%20to%20method%20overloads%20(no%20parenthesis)%2C).